fnl/godbolt/assembly.fnl

```diff
@@ -34,19 +34,90 @@
                 (table.maxn (. map source-buf))
                 (api.nvim_create_buf false true))]
     (api.nvim_buf_set_option buf :filetype :asm)
+    (tset (. vim.bo buf) :bufhidden :unload)
     (api.nvim_buf_set_lines buf 0 -1 true
                             (vim.split text "\n" {:trimempty true}))
     (api.nvim_buf_set_name buf name)
     buf))
 
+(fn get-current-line []
+  (second (fun.getcurpos)))
+
+(fn find-source [asm-buffer]
+  (each [source-buffer asm-buffers (pairs map)]
+    (when (->> asm-buffer (. asm-buffers) (not= nil))
+      (lua "return source_buffer"))))
```

Comment:
Sorry, I've just learned about fennel yesterday. I need to return the first matching value and I haven't managed to do this via `&until`

Reply:
> use `&until` instead of a lua block: https://fennel-lang.org/reference#each-general-iteration

The docs say that `each` returns `nil`. I may do this via `collect`, returning the first element, but it would be unreadable. I'm sure there has to be some canonical way to find something in a table.

Reply:
```fennel
(fn find-source [asm-buffer]
  (do
    (var source-buffer-ret nil)
    (each [source-buffer asm-buffers (pairs map) &until source-buffer-ret]
      (when (. asm-buffers asm-buffer)
        (set source-buffer-ret source-buffer)))
    source-buffer-ret))

```

you also don't have to compare `(. asm-buffers asm-buffer)` with `nil`, `nil` is falsey

Reply:
Done https://github.com/p00f/godbolt.nvim/pull/22/commits/0d3b09db70033cd30a17da85431cb5f041d06efd